app/components/primer/icon_button.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use IconButton to render Icon-only buttons without the default button styles.
+  #
+  # @accessibility
+  #   IconButton requires a `label` which will set the element's `aria-label`, providing assistive technologies with an accessible label.
+  class IconButton < Primer::Component
+    DEFAULT_SCHEME = :default
+    SCHEME_MAPPINGS = {
+      DEFAULT_SCHEME => "",
+      :danger => "btn-octicon-danger"
+    }.freeze
+    SCHEME_OPTIONS = SCHEME_MAPPINGS.keys
+    # @example Default
+    #
+    #   <%= render(Primer::IconButton.new(icon: :search, label: "Search")) %>
+    #
+    # @example Schemes
+    #
+    #   <%= render(Primer::IconButton.new(icon: :search, label: "Search")) %>
+    #   <%= render(Primer::IconButton.new(icon: :trash, label: "Delete", scheme: :danger)) %>
+    #
+    # @param scheme [Symbol] <%= one_of(Primer::IconButton::SCHEME_OPTIONS) %>
+    # @param icon [String] Name of [Octicon](https://primer.style/octicons/) to use.
+    # @param label [String] String that will be read to screenreaders when the component is focused
+    # @param tag [Symbol] <%= one_of(Primer::BaseButton::TAG_OPTIONS) %>
+    # @param type [Symbol] <%= one_of(Primer::BaseButton::TYPE_OPTIONS) %>
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    def initialize(scheme: DEFAULT_SCHEME, icon:, label:, **system_arguments)
```

Comment:
For API consistency, I wonder if we should be surfacing the aria argument that's listed in [system_arguments](https://primer.style/view-components/system-arguments#html-attributes) rather than adding a new keyword (though I do see the benefit of mandating it like this!). 

I think that if it's important to set an `aria-label` for the component, we can mention it in the docs and examples, but otherwise it seems like it wouldn't get set via `**system_arguments`.

Reply:
I agree with @khiga8 comment 👍🏻. Having a `label` argument that doesn't set the `label` attribute is in conflict with the expectations set by `system_arguments`, which wrap `content_tag`.

Reply:
I'd like to discuss a way of **requiring** the aria-label then. In this case, it is pretty hard to come up with a default, since we can use any icon which could mean anything.
what do you think of `raise ArgumentError if @system_arguments[:"aria-label].nil?` at the end of the `initialize`?

Reply:
I think that we do need to require it here, yes. I'd just want to make sure we don't end up with code path that could raise exceptions in production.

Reply:
we can still follow the "don't raise in prod" strategy